kgforge/specializations/stores/bluebrain_nexus.py

```diff
@@ -351,21 +351,38 @@ def retrieve(
                 url, params=query_params, headers=self.service.headers
             )
             response.raise_for_status()
+        except HTTPError as er:
+            nexus_path = f"{self.service.endpoint}/resources/"
+            if not cross_bucket:
+                nexus_path += f"{self.service.organisation}/{self.service.project}"
+            # Try to use the id as it was given
+            if url_base in id_without_query or nexus_path in id:
+                # Use the given id in the request
+                url = id_without_query
+                try:
+                    response = requests.get(
+                        url, params=query_params, headers=self.service.headers
+                    )
+                    response.raise_for_status()
+                except HTTPError as e:
+                    raise RetrievalError(_error_message(e))
+            else:
+                raise RetrievalError(_error_message(er))
+        finally:
             if retrieve_source and not cross_bucket:
 
                 response_metadata = requests.get(
                     url_resource, params=query_params, headers=self.service.headers
```

Comment:
I added `url_resource = id_without_query` instead, because if not it could break the case where the id was given.

Reply:
> because if not it could break the case where the id was given.
Works fine when I try. 
If `id_without_query ` is not used here then:
* forge.retrieve(id=_self, cross_bucket=False, retrieve_source=True) will not retrieve any _store_metadata because /source is always used in try, except and finally clauses. Actually we should always have two calls once without /source and one with to get _store_metadata back.

Reply:
I meant when you do forge.retrieve(id=id) you need the url_resource value, not the id_without_query, that is why I assign url_resource = id_without_query in the except case.